src/core/or/circuituse.c

```diff
@@ -73,6 +73,34 @@
 STATIC void circuit_expire_old_circuits_clientside(void);
 static void circuit_increment_failure_count(void);
 
+/* Return true iff the circuit purpose is for an onion service excluding
+ * client, only service side. */
+static int
+circuit_purpose_is_hs_service(uint8_t purpose)
+{
+  if (purpose == CIRCUIT_PURPOSE_HS_VANGUARDS) {
+    return 1;
+  }
+
+  /* Service-side purpose */
+  if (purpose >= CIRCUIT_PURPOSE_S_HS_MIN_ &&
+      purpose <= CIRCUIT_PURPOSE_S_HS_MAX_) {
+    return 1;
+  }
+
+  return 0;
+}
+
+/* Return true iff the circuit purpose is for an onion service client
+ * excluding service side. */
+static int
+circuit_purpose_is_hs_client(uint8_t purpose)
+{
+  /* Client-side purpose */
```

Comment:
Hmm, the split is not entirely true because vanguards also can be used for clients. There is no error introduced by this commit, but there is a semantic issue. I don't have a strong opinion on how to resolve this.

Reply:
What should I do here hehe? I'm not sure. The `circuit_purpose_is_hs_service()` does look for vanguard also. If `is_hs_client()` can also return true for the Vanguard purpose, we have a problem since both would return true?...

Want me to add a comment saying <insert something> ?

Reply:
Hmm, how about we return True in `is_hs_clietn()` for vanguard circuits, so that in the future if someone uses it, we have taken it into account (and maybe also add it in the func docs)?

Do you think there will be a problem because they both would return true? Why?

Reply:
Unsure if I think there is a problem. I guess if the circuit is an HS `Vanguard` purpose, then it doesn't matter if both return True since all we want to know is if "it is HS" since these function are primarily used in `circuit_purpose_is_hidden_service()`.

Fixup commit has the fix: ec801f4af2ae14f1